controllers/util/solr_util.go

```diff
@@ -452,8 +452,15 @@ func GenerateStatefulSet(solrCloud *solr.SolrCloud, solrCloudStatus *solr.SolrCl
 	}
 
 	if solrCloud.Spec.SolrImage.ImagePullSecret != "" {
-		stateful.Spec.Template.Spec.ImagePullSecrets = []corev1.LocalObjectReference{
-			{Name: solrCloud.Spec.SolrImage.ImagePullSecret},
+		if nil != customPodOptions {
```

Comment:
Just a minor suggestion. I always prefer nil to be on the right, it's just more natural 'if customPodOptions is nothing' as opposed to 'if nothing is customPodOptions'